src/compare-tree/index.js

```diff
@@ -1,28 +1,31 @@
+/**
+ * @flow
+ * @private
+ */
+
 import Node from './node'
 import Query from './query'
-import { getKeyString } from '../key-path'
+import { getKeyString, type KeyPath } from '../key-path'
 
 // The root key is an empty string. This can be a little
 // counter-intuitive, so we keep track of them as a named constant.
 const ROOT_PATH = ''
 
 class CompareTree {
-  /**
-   * @param {*} initial Starting state
-   */
-  constructor(initial) {
+  snapshot: Object
+  nodes: { [key: string]: Node }
+  queries: { [key: string]: Query }
+
+  constructor(initial: Object) {
     this.snapshot = initial
     this.nodes = {}
+    this.queries = {}
```

Comment:
I split the `Query` type into a separate map. This got me really deep flow coverage for a few cases. Still, I think it would be nice to eventually have `Node` and `Query` work from a common interface, if possible.